librz/include/rz_arch.h

```diff
@@ -59,6 +61,245 @@ RZ_API bool rz_arch_load_platform_sdb(RZ_NONNULL RzArchPlatformTarget *t, RZ_NON
+	RZ_ARCH_OPCODE_TYPE_NOP /*                   */ = (1ull << 1), ///< Opcode performs a no-operation instruction
+
+	/* Branch Instructions */
+	RZ_ARCH_OPCODE_TYPE_JUMP /*                  */ = (1ull << 2), ///< Opcode performs a jump to a known location (sets info.reference.[absolute|relative]_jump)
+	RZ_ARCH_OPCODE_TYPE_JUMP_UNKNOWN /*          */ = (1ull << 3), ///< Opcode performs a jump to an unknown location (register jump, indirect jump, etc...)
+	RZ_ARCH_OPCODE_TYPE_CALL /*                  */ = (1ull << 4), ///< Opcode performs a call a subroutine to a known location (sets info.reference.[absolute|relative]_jump)
+	RZ_ARCH_OPCODE_TYPE_CALL_UNKNOWN /*          */ = (1ull << 5), ///< Opcode performs a call a subroutine to an unknown location (register jump, indirect jump, etc...)
+	RZ_ARCH_OPCODE_TYPE_RETURN /*                */ = (1ull << 6), ///< Opcode performs a return from a subroutine
+
+	/* Data Transfer Instructions */
+	RZ_ARCH_OPCODE_TYPE_LOAD /*                  */ = (1ull << 7), ///< Opcode performs a load operation (i.e. load a value from the memory)
+	RZ_ARCH_OPCODE_TYPE_STORE /*                 */ = (1ull << 8), ///< Opcode performs a store operation (i.e. store a value to the memory)
+	RZ_ARCH_OPCODE_TYPE_PUSH /*                  */ = (1ull << 9), ///< Opcode performs a push operation (i.e. push value to stack)
+	RZ_ARCH_OPCODE_TYPE_POP /*                   */ = (1ull << 10), ///< Opcode performs a pop operation (i.e. pop value to stack)
+
+	/* Compare Instructions */
+	RZ_ARCH_OPCODE_TYPE_COMPARE /*               */ = (1ull << 11), ///< Opcode performs a compare operation
+	RZ_ARCH_OPCODE_TYPE_TEST /*                  */ = (1ull << 12), ///< Opcode performs a tests operation (i.e. `if !(a & b)` operation)
+
+	/* Cast Instructions */
+	RZ_ARCH_OPCODE_TYPE_ZERO_EXTEND /*           */ = (1ull << 13), ///< Opcode performs a extend zero operation (cast to unsigned)
+	RZ_ARCH_OPCODE_TYPE_SIGN_EXTEND /*           */ = (1ull << 14), ///< Opcode performs a extend sign operation (cast to signed)
+
+	/* Mathematical Instructions */
+	RZ_ARCH_OPCODE_TYPE_ADD /*                   */ = (1ull << 15), ///< Opcode performs a addition (a + b)
+	RZ_ARCH_OPCODE_TYPE_SUBTRACT /*              */ = (1ull << 16), ///< Opcode performs a subtraction (a − b)
+	RZ_ARCH_OPCODE_TYPE_MULTIPLY /*              */ = (1ull << 17), ///< Opcode performs a multiplication (a * b)
+	RZ_ARCH_OPCODE_TYPE_DIVIDE /*                */ = (1ull << 18), ///< Opcode performs a division (b / a)
+	RZ_ARCH_OPCODE_TYPE_MODULO /*                */ = (1ull << 19), ///< Opcode performs a modulus, i.e. remainder of a division (b % a)
+	RZ_ARCH_OPCODE_TYPE_NEGATE /*                */ = (1ull << 20), ///< Opcode performs a negate operation, i.e inverts the sign of a value (a = -b)
+	RZ_ARCH_OPCODE_TYPE_MATH /*                  */ = (1ull << 21), ///< Opcode performs a math operation not described above (i.e. sqrt, log2, abs, etc...)
```

Comment:
We probably also should distinguish between Integer math and Real math instructions, especially in the light of RZIL.